pyflow/host.py

```diff
@@ -186,21 +186,52 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    @property
-    def ecflow_variables(self):
-        """*dict*: The variables that must be set on relevant nodes to run on this host."""
+    def update_node_attributes(self, options):
+        """
+        Updates the attributes of a node with the host-specific values.
+
+        Parameters:
+        - options (dict): The options dictionary to update with host-specific attributes.
+        """
         if self.server_ecfvars:
-            vars = {}
+            # Use generated variables to be able to export the variables in tasks
+            host_attrs = {
+                "generated_variables": [
+                    "ECF_JOB_CMD",
+                    "ECF_KILL_CMD",
+                    "ECF_STATUS_CMD",
+                    "ECF_CHECK_CMD",
+                    "ECF_OUT",
+                ],
+                "variables": {**self.extra_variables},
+            }
         else:
-            vars = {
-                "ECF_JOB_CMD": self.job_cmd,
-                "ECF_KILL_CMD": self.kill_cmd,
-                "ECF_STATUS_CMD": self.status_cmd,
-                "ECF_CHECK_CMD": self.check_cmd,
-                "ECF_OUT": self.log_directory,
+            host_attrs = {
+                "generated_variables": [],
+                "variables": {
+                    "ECF_JOB_CMD": self.job_cmd,
+                    "ECF_KILL_CMD": self.kill_cmd,
+                    "ECF_STATUS_CMD": self.status_cmd,
+                    "ECF_CHECK_CMD": self.check_cmd,
+                    "ECF_OUT": self.log_directory,
+                    **self.extra_variables,
+                },
             }
-        vars.update(self.extra_variables)
-        return vars
+
+        # Update the options with host-specific attributes
+        for attribute, values in host_attrs.items():
+            if attribute in ["variables"]:
+                variables = options.pop("variables", {})
```

Comment:
Could we change `"variables"` to `attribute`? Just so it works if we add additional attributes into `host_attrs`. Similar change needed for `"generated_variables"` below.